### PR TITLE
WIP for adding BellSound to schema

### LINKED ISSUE #12504: `bellSound` missing from JSON schema
We should add the `bellSound` profile setting to the JSON schema.

Feature implemented here: https://github.com/microsoft/terminal/pull/11511

🦤🔪

### PR DESCRIPTION
**Pull request by pizzaz93**
## PR Checklist
* [X] Closes #12504
* [X] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [X] Documentation previously added by another contributor https://docs.microsoft.com/en-us/windows/terminal/customize-settings/profile-advanced#bell-sound-preview
* [X] Schema updated.

### CODE REVIEW

**Comment by pizzaz93** [comment]
I'm not sure what the default is and the regex for the pattern?
Additional questions would be about the placement of `BellSound` within the schema and if there are additional fields that should be added.

File: doc/cascadia/profiles.schema.json [hunk 1]
```diff
@@ -54,6 +54,12 @@
         }
       ]
     },
+    "BellSound": {
+      "default": "#",
+      "description": "Controls the sound the Terminal plays when the application emits a BEL character. If the path is invalid no sound will be played. This property also accepts an array of sounds and the Terminal will pick one at random.",
+      "pattern": "",
+      "type": "string",
```
Comment (carlos-zamora) [inline doc/cascadia/profiles.schema.json:61]: I suggest taking a look at how we handle `backgroundImage`...
https://github.com/microsoft/terminal/blob/1aa717a60d9e7cded96b83fda98c88be24adbb94/doc/cascadia/profiles.schema.json#L132-L151

We should be able to accept null, a string, or an array of strings.

File: doc/cascadia/profiles.schema.json [hunk 2]
```diff
@@ -54,6 +54,12 @@
         }
       ]
     },
+    "BellSound": {
+      "default": "#",
+      "description": "Controls the sound the Terminal plays when the application emits a BEL character. If the path is invalid no sound will be played. This property also accepts an array of sounds and the Terminal will pick one at random.",
```
Comment (carlos-zamora) [inline doc/cascadia/profiles.schema.json:59]:
```suggestion
      "description": "Sets the file location of the sound played when the application emits a BEL character. If the path is invalid no sound will be played. This property also accepts an array of sounds and the terminal will pick one at random.",
```

File: doc/cascadia/profiles.schema.json [hunk 3]
```diff
@@ -54,6 +54,12 @@
         }
       ]
     },
+    "BellSound": {
+      "default": "#",
```
Comment (carlos-zamora) [inline doc/cascadia/profiles.schema.json:58]:
```suggestion
      "default": "",
```
I don't thing having the default path be `"#"` would be useful haha. Let's just make it the empty string.

File: doc/cascadia/profiles.schema.json (hunk 1 above)
Comment (carlos-zamora) [reply doc/cascadia/profiles.schema.json:61]: `BellStyle` is another good example to look at. It should show you how to make an array:
https://github.com/microsoft/terminal/blob/1aa717a60d9e7cded96b83fda98c88be24adbb94/doc/cascadia/profiles.schema.json#L34-L44

**Review by carlos-zamora — CHANGES_REQUESTED** [review]

File: doc/cascadia/profiles.schema.json (hunk 1 above)
Comment (pizzaz93) [reply doc/cascadia/profiles.schema.json]: I believe I captured the values correctly.

File: doc/cascadia/profiles.schema.json [hunk 4]
```diff
@@ -54,6 +54,12 @@
         }
       ]
     },
+    "BellSound": {
+      "default": "",
+      "description": "Sets the file location of the sound played when the application emits a BEL character. If the path is invalid no sound will be played. This property also accepts an array of sounds and the terminal will pick one at random.",
+      "pattern": "",
```
Comment (pizzaz93) [inline doc/cascadia/profiles.schema.json:60]: @carlos-zamora would we want regex for a local file or an empty string is alright?

Comment (carlos-zamora) [reply doc/cascadia/profiles.schema.json:60]: eh, empty string is fine. I think that's what we did with `backgroundImage` which is a similar idea.

File: doc/cascadia/profiles.schema.json [hunk 5]
```diff
@@ -54,6 +54,23 @@
...
+      "oneOf": [
+      {
+        "type": "string",
+        "type": "null"
```
Comment (carlos-zamora) [inline doc/cascadia/profiles.schema.json:64]:
```suggestion
        "type": [
          "string",
          "null"
        ]
```
`backgroundImage` has it like this, so I'm gonna suggest that it look like this haha.

File: doc/cascadia/profiles.schema.json [hunk 6]
```diff
@@ -54,6 +54,23 @@
...
+      },
+      {
+        "type": "array",
+          "items": {
+            "type": "string"
+          }
```
Comment (carlos-zamora) [inline doc/cascadia/profiles.schema.json:70]:
```suggestion
        "type": "array",
        "items": {
          "type": "string"
        }
```
nit: just fixing the indentation here.

File: doc/cascadia/profiles.schema.json (hunk 4 above)
Comment (carlos-zamora) [reply doc/cascadia/profiles.schema.json:60]: In fact, I suggest outright removing the pattern property here. I don't think it does anything (or at least anything valuable here?)

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Last few things and I'm ready to hit that ✅! 😊

File: doc/cascadia/profiles.schema.json (hunk 4 above)
Comment (pizzaz93) [reply doc/cascadia/profiles.schema.json]: Removed the pattern variable

File: doc/cascadia/profiles.schema.json [hunk 7]
```diff
@@ -54,6 +54,24 @@
...
+          "string",
+          "null"
+        ]
+      },
+      {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+        }
```
Comment (carlos-zamora) [inline doc/cascadia/profiles.schema.json:72]:
```suggestion
      }
```

Comment (carlos-zamora) [reply doc/cascadia/profiles.schema.json:72]: nit: alignment

**Review by carlos-zamora — APPROVED** [review]
Awesome. Thanks for doing this!

**Comment by pizzaz93** [comment]
> Awesome. Thanks for doing this!

You're very welcome, glad to help.

**Review by DHowett — APPROVED** [review]
Thank you!

**Comment by DHowett** [comment]
@msftbot merge this in 2 minutes

**Comment by DHowett** [comment]
Oh, it is a schema update. The bot will never merge it.

**Pull request merged**